event-adapter/cmd/provider/provider.go

```diff
@@ -14,363 +14,49 @@ limitations under the License.
-	err := apimeta.EachListItem(list, func(item runtime.Object) error {
-		objMeta := item.(metav1.ObjectMetaAccessor).GetObjectMeta()
-		resourceIDs = append(resourceIDs, fmt.Sprintf("\"%s\"", objMeta.GetUID()))
-		return nil
-	})
-	if err == nil {
-		glog.Infof("resource names: ", resourceIDs)
-	}
-	return resourceIDs, err
-}
-
-// STACKDRIVER IO FUNCTIONS
-// TODO: extract to anoter file
-
-type ListPodMetricRequest struct {
-	project string
-	cluster_name string
-	zone string
-	pod_ids []string
-  metricName string
-	interval time.Duration
-}
-
-func createMetricFilter(request *ListPodMetricRequest) (string, error) {
-	// project_id, cluster_name and pod_id together identify a pod unambiguously
-	projectFilter := fmt.Sprintf("resource.label.project_id = %s", request.project)
-	zoneFilter := fmt.Sprintf("resource.label.zone = %s", request.zone)
-	clusterFilter := fmt.Sprintf("resource.label.cluster_name = %s", request.cluster_name)
-	// container_name is set to empty string for pod metrics
-	containerFilter := "resource.label.container_name = \"\""
-	metricFilter := fmt.Sprintf("metric.type = \"%s\"", request.metricName)
-
-	var nameFilter string
-	if (request.pod_ids == nil) {
-		return fmt.Sprintf("(%s) AND (%s) AND (%s) AND (%s) AND (%s)", metricFilter, containerFilter, clusterFilter, projectFilter, zoneFilter), nil
-	} else if len(request.pod_ids) == 0 {
-		glog.Infof("no pods matched for metric %s", request.metricName)
-		return "", fmt.Errorf("no pods matched")
-	} else if len(request.pod_ids) == 1 {
-		nameFilter = fmt.Sprintf("resource.label.pod_id = %s", request.pod_ids[0])
-	} else {
-		nameFilter = fmt.Sprintf("resource.label.pod_id = one_of(%s)", strings.Join(request.pod_ids, ","))
-	}
-	return fmt.Sprintf("(%s) AND (%s) AND (%s) AND (%s) AND (%s) AND (%s)", metricFilter, nameFilter, containerFilter, clusterFilter, projectFilter, zoneFilter), nil
-}
-
-func (s *sdService) ListPodMetric(request *ListPodMetricRequest) (*stackdriver.ListTimeSeriesResponse, error) {
-	project := fmt.Sprintf("projects/%s", request.project)
-	endTime := time.Now()
-	startTime := endTime.Add(-request.interval)
-	filter, err := createMetricFilter(request)
-	if err != nil {
-		return nil, err
-	}
-	sdRequest := s.Projects.TimeSeries.List(project).Filter(filter).
-			IntervalStartTime(startTime.Format("2006-01-02T15:04:05Z")).
-			IntervalEndTime(endTime.Format("2006-01-02T15:04:05Z")).
-			AggregationPerSeriesAligner("ALIGN_MEAN").
-			AggregationAlignmentPeriod(fmt.Sprintf("%vs", int64(request.interval.Seconds())))
-	glog.Infof("request following: %s", sdRequest)
-	return sdRequest.Do()
-}
-
-func (s *sdService) ListAllMetrics(project string, prefix string) (*stackdriver.ListMetricDescriptorsResponse, error) {
-	onlyCustom := fmt.Sprintf("metric.type = starts_with(\"%s/\")", prefix)
-	return s.Projects.MetricDescriptors.List(fmt.Sprintf("projects/%s", project)).Filter(onlyCustom).Do()
+// GetNamespacedEventsByName get the information of the given event
```

Comment:
Fix your comments: "creates" instead of "create", "gets" instead of "get".

I'd prefer a more accurate error message, for example: "GetNamespacedEventsByName is not implemented yet"